projects/optic/src/commands/oas/capture.ts

```diff
@@ -203,62 +245,65 @@ export async function captureCommand(config: OpticCliConfig): Promise<Command> {
         renderingStats,
         trackingStats,
       ]);
+      const complete = async () =>
+        Promise.all([
+          !runningCommand ? await systemProxy.stop() : Promise.resolve(),
+          (async () => {
+            if (options.output) {
+              const outputPath = path.resolve(options.output);
+              feedback.success(`Wrote har to ${outputPath}`);
+              return await fs.rename(inProgressName, outputPath);
+            }
+            await fs.rename(inProgressName, completedName);
+
+            try {
+              await runVerify(
+                filePath,
+                {
+                  exit0: true,
+                  har: options.output
+                    ? path.resolve(options.output)
+                    : undefined,
+                },
+                config,
+                feedback,
+                {
+                  printCoverage: false,
+                }
+              );
+            } catch (e) {
+              console.log(e);
+            }
 
-      exitHook((callback) => {
-        sourcesController.abort();
+            if (!interactiveCapture) {
+              // Log next steps
+              feedback.success(`Wrote har traffic to ${completedName}`);
+              feedback.log(
+                `\nRun "${chalk.bold(
+                  `optic verify ${filePath}`
+                )}" to diff the captured traffic`
+              );
+            }
+          })(),
+        ]);
 
-        completing
-          .then(async () =>
-            Promise.all([
-              !runningCommand ? await systemProxy.stop() : Promise.resolve(),
-              (async () => {
-                if (options.output) {
-                  const outputPath = path.resolve(options.output);
-                  feedback.success(`Wrote har to ${outputPath}`);
-                  return await fs.rename(inProgressName, outputPath);
-                }
-                await fs.rename(inProgressName, completedName);
-
-                try {
-                  await runVerify(
-                    filePath,
-                    {
-                      exit0: true,
-                      har: options.output
-                        ? path.resolve(options.output)
-                        : undefined,
-                    },
-                    config,
-                    feedback,
-                    {
-                      printCoverage: false,
-                    }
-                  );
-                } catch (e) {
-                  console.log(e);
-                }
+      await completing;
+      if (serverRunner) serverRunner.kill();
+      if (!interactiveCapture) {
+        await complete();
+        process.exit(exitCode ?? 0);
```

Comment:
This was kind of weird and I'm not entirely sure what's happening here - we're not properly tearing down something in our spawn process (`--command` and `--server` are both affected)

Using the hammer approach to just exit in non-interactive mode, but we should look into what's causing this